### PR TITLE
Ignore `--test` flag when it is an user-provided argument

### PR DESCRIPTION
**Pull request by Dowsley**
Simple fix for edge-case https://github.com/godotengine/godot/issues/95922

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -861,6 +861,10 @@ void Main::test_cleanup() {
 
 int Main::test_entrypoint(int argc, char *argv[], bool &tests_need_run) {
 	for (int x = 0; x < argc; x++) {
+		// Early return to ignore a possible user-provided "--test" argument.
+		if ((strlen(argv[x]) == 2) && ((strncmp(argv[x], "--", 2) == 0) || (strncmp(argv[x], "++", 2) == 0))) {
+			return EXIT_SUCCESS;
+		}
```
Comment (Calinou) [inline main/main.cpp:867]: Set `tests_need_run` to `false` as is done when not specifying `--test` near the end of the function:

```suggestion
		if ((strlen(argv[x]) == 2) && ((strncmp(argv[x], "--", 2) == 0) || (strncmp(argv[x], "++", 2) == 0))) {
			tests_need_run = false;
			return EXIT_SUCCESS;
		}
```

**Review by Calinou — COMMENTED** [review]
Tested locally (rebased on top of `master` https://github.com/godotengine/godot/commit/77d6283d22731398e23c3f1e1b4eeedd22a93152), it works as expected.

Should be good to merge after applying suggestions.

**Comment by Calinou** [comment]
Looks good to me, please squash the commits together so we can merge this. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for instructions :slightly_smiling_face:

**Comment by Dowsley** [comment]
> Looks good to me, please squash the commits together so we can merge this. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for instructions 🙂

Since the change was so small I remade the branch but sitting on top of current master. Force pushed.

**Review by Calinou — APPROVED** [review]

File: main/main.cpp [hunk 2]
```diff
@@ -889,6 +889,11 @@ void Main::test_cleanup() {
 
 int Main::test_entrypoint(int argc, char *argv[], bool &tests_need_run) {
 	for (int x = 0; x < argc; x++) {
+		// Early return to ignore a possible user-provided "--test" argument.
+		if ((strlen(argv[x]) == 2) && ((strncmp(argv[x], "--", 2) == 0) || (strncmp(argv[x], "++", 2) == 0))) {
+			tests_need_run = false;
+			return EXIT_SUCCESS;
+		}
```
Comment (akien-mga) [inline main/main.cpp:896]: Wouldn't this make `godot --test -- --yolo` fail to run tests?

**Comment by akien-mga** [comment]
This is getting somewhat hacky for a niche use case, I might be more in favor of reverting #90507 if this has such implications.

File: main/main.cpp (hunk 2 above)
Comment (akien-mga) [reply main/main.cpp:896]: I guess not, this will process in order so `--test` would be seen and handled first.

**Review by akien-mga — APPROVED** [review]
Had a quick look at improving this, but indeed `--test` gets processed before the regular argument parsing so it's naturally a bit hardcoded.

Should be fine for now, I might poke at it later to make this clear as I'm not fond of this `TEST_MAIN_OVERRIDE`.

File: main/main.cpp (hunk 2 above)
Comment (Dowsley) [reply main/main.cpp:896]: Exactly. But I agree that this is hacky.

Won't https://github.com/godotengine/godot/pull/90507 be missed though?

Comment (Dowsley) [reply main/main.cpp:896]: We could rewrite that early return specifically.

Comment (akien-mga) [reply main/main.cpp:896]: I'm not sure how to make it better, in the end I concluded that it's probably good enough for now.

**Comment by Repiteo** [comment]
Thanks!
